src/main/java/org/elasticsearch/index/query/QueryFilterBuilder.java

```diff
@@ -29,47 +29,29 @@
  *             query as a filter directly.
  */
 @Deprecated
-public class QueryFilterBuilder extends QueryBuilder {
+public class QueryFilterBuilder extends QueryBuilder<QueryFilterBuilder> {
 
-    private final QueryBuilder queryBuilder;
+    public static final String NAME = "query";
 
-    private String queryName;
+    private final QueryBuilder<?> queryBuilder;
 
     /**
      * A filter that simply wraps a query.
      *
      * @param queryBuilder The query to wrap as a filter
      */
-    public QueryFilterBuilder(QueryBuilder queryBuilder) {
+    public QueryFilterBuilder(QueryBuilder<?> queryBuilder) {
         this.queryBuilder = queryBuilder;
     }
 
-    /**
-     * Sets the query name for the filter that can be used when searching for matched_filters per hit.
-     */
-    public QueryFilterBuilder queryName(String queryName) {
-        this.queryName = queryName;
-        return this;
-    }
-
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
-        if (queryName == null) {
-            builder.field(QueryFilterParser.NAME);
-            queryBuilder.toXContent(builder, params);
-        } else {
-            builder.startObject(FQueryFilterParser.NAME);
-            builder.field("query");
+            builder.field(NAME);
             queryBuilder.toXContent(builder, params);
-            if (queryName != null) {
-                builder.field("_name", queryName);
-            }
-            builder.endObject();
-        }
```

Comment:
All right, will remove FQueryFilterBuilder then, put the NAME of the extra FQueryFilterParser somewhere else in the QueryFilterBuilder then.